src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -285,28 +282,18 @@ protected void AssertSelf ()
 					ret.RequestedAuthentication = RequestedAuthentication;
 					return ret;
 			}
-
+			
 			if (!IsErrorStatusCode (statusCode)) {
 				if (Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, $"Reading...");
-				Stream inputStream = new BufferedStream (httpConnection.InputStream);
-				if (decompress_here) {
-					string[] encodings = httpConnection.ContentEncoding?.Split (',');
-					if (encodings != null) {
-						if (encodings.Contains (GZIP_ENCODING, StringComparer.OrdinalIgnoreCase))
-							inputStream = new GZipStream (inputStream, CompressionMode.Decompress);
-						else if (encodings.Contains (DEFLATE_ENCODING, StringComparer.OrdinalIgnoreCase))
-							inputStream = new DeflateStream (inputStream, CompressionMode.Decompress);
-					}
-				}
-				ret.Content = new StreamContent (inputStream);
-			} else {
+				ret.Content = GetContent(httpConnection, httpConnection.InputStream);
+			}
+			else {
 				if (Logger.LogNet)
-					Logger.Log (LogLevel.Info, LOG_APP, $"Status code is {statusCode}, returning empty content");
-				// For 400 >= response code <= 599 the Java client throws the FileNotFound exeption when attempting to read from the connection
-				// Client tests require we return no content here
```

Comment:
Please restore the comment in appropriate place, it contains actual useful information for the coder